components/brave_rewards/resources/extension/brave_rewards/background/events/rewardsEvents.ts

```diff
@@ -4,10 +4,23 @@
 
 import rewardsPanelActions from '../actions/rewardsPanelActions'
 
-chrome.braveRewards.getAllNotifications((list: RewardsExtension.Notification[]) => {
-  rewardsPanelActions.onAllNotifications(list)
+// Fetch initial data required to refresh state, keeping in mind
+// that the extension process be restarted at any time.
+// TODO(petemill): Move to initializer function or single 'init' action.
+chrome.braveRewards.getRewardsMainEnabled((enabledMain: boolean) => {
+  rewardsPanelActions.onEnabledMain(enabledMain)
+  if (enabledMain) {
+    chrome.braveRewards.getWalletProperties()
+    chrome.braveRewards.fetchBalance((balance: RewardsExtension.Balance) => {
+      rewardsPanelActions.onBalance(balance)
+    })
+    chrome.braveRewards.getAllNotifications((list: RewardsExtension.Notification[]) => {
+      rewardsPanelActions.onAllNotifications(list)
+    })
+  }
 })
```

Comment:
@NejcZdovc @ryanml can you verify that thihs is all the data that the extension background script will need if it is started up sometimes _after_ rewards is enabled? With this PR, it's started _very_ quickly after being enabled, but when we change to being non-persistent extension, it can lose it's memory at any time in-between handling events. Of course more will need to change when we do that. So for now can you verify that this is enough data fetching to handle this case.

Reply:
@petemill I think we should also fetch recurring tip data like here: https://github.com/brave/brave-core/blob/master/components/brave_rewards/resources/extension/brave_rewards/components/app.tsx#L57

this will set the current recurring tip amount or prompt the user to set a recurring monthly tip if they have no recurring contributions to the publisher, this should be accurate in the panel as soon as possible